compiler/test/stdlib/int16.test.gr

```diff
@@ -0,0 +1,138 @@
+module Int16Test
+
+include "int16"
+from Int16 use {
+  fromNumber,
+  toNumber,
+  fromUint16,
+  incr,
+  decr,
+  lnot,
+  rem,
+  (+),
+  (-),
+  (*),
+  (/),
+  (%),
+  (&),
+  (|),
+  (^),
+  (<<),
+  (>>),
+}
+
+// Suppress warnings about using `fromNumber` on constants, since that's what we want to test.
+let fromNumber = fromNumber
+
+assert fromNumber(5) == 5S
```

Comment:
Possible consideration: `S` looks pretty similar to `5`, which may make distinguishing `Int16` literals from normal numbers ending in `5` a bit hard. Should the literal notation be changed or is it fine keeping it like this?

Reply:
Nah, I think it's fine. You can always add an underscore for readability; `5_S`.

Reply:
We can also highlight it better in the LSP grammar. https://github.com/grain-lang/grain-language-server/issues/148